db/migrate/20200626100128_change_activity_parent_relationship_name.rb

```diff
@@ -0,0 +1,23 @@
+class ChangeActivityParentRelationshipName < ActiveRecord::Migration[6.0]
+  def up
+    add_column :activities, :parent_id, :string
+    Activity.transaction do
+      Activity.all.each do |activity|
+        activity.update!(parent_id: activity.activity_id)
```

Comment:
We shold handle the case where an activity is incomplete - as discussed setting `validate: false` is a nice clear wat to do that.